src/obstacle_params_parser.cpp

```diff
@@ -0,0 +1,133 @@
+#pragma once
+
+#include "panda_torque_mpc/obstacle_params_parser.h"
+
+#include <Eigen/Dense>
+#include <boost/shared_ptr.hpp>
+#include <fstream>
+#include <memory>
+#include <vector>
+
+#include <pinocchio/fwd.hpp>
+#include <pinocchio/multibody/model.hpp>
+
+#include <ros/ros.h>
+#include <yaml-cpp/yaml.h>
+
+namespace panda_torque_mpc {
+
+  ObstacleParamsParser::ObstacleParamsParser(
+      const std::shared_ptr<ros::NodeHandle> &nh,
+      const boost::shared_ptr<pinocchio::Model> &pin_model,
+      const boost::shared_ptr<pinocchio::GeometryModel> &collision_model)
+      : nh_(nh), pin_model_(pin_model), collision_model_(collision_model){};
+
+  ObstacleParamsParser::~ObstacleParamsParser() {};
+
+void ObstacleParamsParser::addCollisions()
+{
+  // First adding the obstacles to the geometry model
+ int obs_idx = 0;
+ while (nh_->hasParam("obstacle_" + std::to_string(obs_idx))) {
+
+  std::string obstacle_name;
+
+  obstacle_name = "obstacle_" + std::to_string(obs_idx);
+  std::string type;
+
+  nh_->getParam(obstacle_name + "/type", type);
```

Comment:
```suggestion
  std::string obstacle_name;
  obstacle_name = "obstacle_" + std::to_string(obs_idx);
  
  std::string type;
  nh_->getParam(obstacle_name + "/type", type);
```
Formatting the code like this better indicates which string declaration belongs to which operation